File: app/src/main/java/com/example/daniel/robotspencilscodetest/view/CommentContract.java
package com.example.daniel.robotspencilscodetest.view;

import com.example.daniel.robotspencilscodetest.*;


/**
 * Created by Daniel on 2/8/17.
 */

public interface CommentContract {
    interface View extends BaseView{
        void showData(String data);
    }

    interface Presenter extends BasePresenter<View>{
        void loadData();
    }
}

File: app/src/main/java/com/example/daniel/robotspencilscodetest/view/CommentContract.kt
package com.example.daniel.robotspencilscodetest.view

import com.example.daniel.robotspencilscodetest.BasePresenter
import com.example.daniel.robotspencilscodetest.BaseView

/**
 * Created by Daniel on 4/20/17.
 */
interface CommentContract {

    //User action
    interface Presenter : BasePresenter<CommentContract.View> {
        fun loadComments()
        fun pushToFirebase();
    }

    //Action callback
    interface View : BaseView {
        fun onTextLoaded(text : String)
    }
}